src/identity/caclient.rs

```diff
@@ -217,6 +218,9 @@ pub mod mock {
             let not_after = not_before + self.cfg.cert_lifetime;
 
             let mut state = self.state.write().await;
+            if state.error {
```

Comment:
Ditto - this can be in a conditionally-compiled block, which both serves to document why the logic is here, and is safer.

Reply:
the entire block is conditionally compile (`mod mock`)

Reply:
Ugh, I didn't expand everything. My bad.